ReactApp/src/AppRoutes.jsx

```diff
@@ -0,0 +1,327 @@
+import UserProfile from "React-Automation-Studio/components/SystemComponents/userProfiles/UserProfile";
+import ProtectedRoute from "React-Automation-Studio/components/SystemComponents/ProtectedRoute";
+
+const AppRoutes = (props) => {
+  const context = useContext(AutomationStudioContext);
+  /* eslint-disable-next-line no-unused-vars */
+  const userData = context.userData;
+  /* eslint-disable-next-line no-unused-vars */
+  const roles = context.userData.roles;
+  /* eslint-disable-next-line no-unused-vars */
+  const username = context.userData.username;
+  let enableDemos =
+    typeof import.meta.env.VITE__DISABLE_DEMOS !== "undefined"
+      ? !(import.meta.env.VITE__DISABLE_DEMOS.toUpperCase() === "TRUE")
+      : true;
+
+  return (
+    <BrowserRouter>
+      <Routes>
+      <Route
+          exact
+          path="/"
+          element={
+            <ProtectedRoute>
+              <AppMain/>
+            </ProtectedRoute>
+          }
+        />
+       <Route
+          exact
+          path="/MainDashboard"
+          element={
+            <ProtectedRoute>
+              <MainDashboard />
+            </ProtectedRoute>
+          }
+        />
+        <Route
+          exact
+          path="/Administrator"
+          element={
+            <ProtectedRoute roles={["admin"]}>
+              <Administrator />
+            </ProtectedRoute>
+          }
+        />
+        {/* <ProtectedRoute path="/UserProfile" component={UserProfile} /> */}
+        {import.meta.env.VITE__EnableLogin === "true" && (
+          <Route
+            exact
+            path="/Login"
+            component={() => (
+              <Login
+                version="V4.0.3"
+                timeout={5000}
+              />
+            )}
+          />
+        )}
+        <Route
+          path="/UserProfile"
+          element={
+            <ProtectedRoute>
+              <UserProfile />
+            </ProtectedRoute>
+          }
+        />
+        {import.meta.env.VITE_EnableLogin === "true" && (
+          <Route
+            exact
+            path="/Login"
+            element={<Login version="V5.0.0" timeout={5000} />}
+          />
+        )}
+
+        <Route
+          path="/Probe"
+          element={
+            <ProtectedRoute>
+              <Probe />
+            </ProtectedRoute>
+          }
+        />
+        <Route
+          path="/Help"
+          element={
+            <ProtectedRoute>
+              <Help />
+            </ProtectedRoute>
+          }
+        />
+        {/*system end*/}
+
+        {/*demos start*/}
+        {enableDemos && (
+          <Route
+          exact
+          path="/DemosDashboard"
+          element={
+            <ProtectedRoute>
+              <MainDashboard />
+            </ProtectedRoute>
+          }
+        />
+        )}
+        {/* {enableDemos && <ProtectedRoute path="/Demos" component={Demos} />} */}
```

Comment:
Can we add some comment to explain the purpose (e.g. "For restricted access use: ....").